File: base-v3/src/main/java/com/adyen/checkout/base/component/lifecycle/BaseLifecycleObserver.java
/*
 * Copyright (c) 2020 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 8/9/2020.
 */

package com.adyen.checkout.base.component.lifecycle;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;

/**
 * Basic LifecycleObserver that can be extended.
 */
public class BaseLifecycleObserver implements LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    public void onCreate(){
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    public void onStart(){
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    public void onResume(){
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    public void onPause(){
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    public void onStop(){
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    public void onDestroy(){
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    public void onAny(){
        // noop, to be overridden
    }
}

File: components-core/src/main/java/com/adyen/checkout/components/base/lifecycle/BaseLifecycleObserver.kt
/*
 * Copyright (c) 2020 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 8/9/2020.
 */
package com.adyen.checkout.components.base.lifecycle

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent

/**
 * Basic LifecycleObserver that can be extended.
 */
open class BaseLifecycleObserver : LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate() {
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart() {
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    open fun onResume() {
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() {
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onStop() {
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy() {
        // noop, to be overridden
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    fun onAny() {
        // noop, to be overridden
    }
}
